samwise.js

```diff
@@ -11,6 +11,7 @@
 
```

Comment:
SCRIPT-Consider extracting this repeated logic into a separate function to avoid duplication.